modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js

```diff
@@ -48,53 +49,107 @@ import {paginationReducer} from './reducers/index.es';
 			event.preventDefault();
 
-			apiRef.current
-				.validate()
-				.then((validForm) => {
-					if (validForm) {
-						const liferayForm =
-							event.target.id &&
-							Liferay.Form.get(event.target.id);
-
-						const validLiferayForm = !Object.keys(
-							liferayForm?.formValidator?.errors ?? {}
-						).length;
-
-						if (!validLiferayForm) {
-							return;
-						}
-
-						if (submittable) {
-							Liferay.Util.submitForm(event.target);
-						}
-
-						Liferay.fire('ddmFormValid', {
-							formWrapperId: event.target.id,
-						});
-
-						Liferay.fire('ddmFormSubmit', {
-							formId: getFormId(
-								getFormNode(containerRef.current)
-							),
-						});
+			fetch(validateCsrfTokenURL, {
+				headers: new Headers({
+					Accept: 'application/json',
+					'Content-Type': 'application/json',
+				}),
+				method: 'GET',
+			})
+				.then((response) => {
+					return response.json();
+				})
+				.then((jsonResponse) => {
+					const validCsrfToken = jsonResponse.validCsrfToken;
+
+					event.preventDefault();
+
+					if (validCsrfToken) {
+						apiRef.current
+							.validate()
+							.then((validForm) => {
+								if (validForm) {
+									const liferayForm =
+										event.target.id &&
+										Liferay.Form.get(event.target.id);
+
+									const validLiferayForm = !Object.keys(
+										liferayForm?.formValidator?.errors ?? {}
+									).length;
+
+									if (!validLiferayForm) {
+										return;
+									}
+
+									if (submittable) {
+										Liferay.Util.submitForm(event.target);
+									}
+
+									Liferay.fire('ddmFormValid', {
+										formWrapperId: event.target.id,
+									});
+
+									Liferay.fire('ddmFormSubmit', {
+										formId: getFormId(
+											getFormNode(containerRef.current)
+										),
+									});
+								}
+								else {
+									Liferay.fire('ddmFormError', {
+										formWrapperId: event.target.id,
+									});
+								}
+							})
+							.catch((error) => {
+								console.error(error);
+							});
 					}
 					else {
-						Liferay.fire('ddmFormError', {
-							formWrapperId: event.target.id,
-						});
+						if (Liferay.ThemeDisplay.isSignedIn()) {
+							openModal({
+								bodyHTML: Liferay.Language.get(
+									'you-need-to-be-signed-in-to-submit-this-form'
+								),
+								buttons: [
+									{
+										displayType: 'secondary',
+										label: Liferay.Language.get('cancel'),
+										type: 'cancel',
+									},
+									{
+										label: Liferay.Language.get('ok'),
+										onClick: () => {
+											location.href =
+												themeDisplay.getPathMain() +
+												'/portal/login?redirect=' +
+												window.location.href;
+										},
+									},
+								],
+								id:
+									'<portlet:namespace />ddmFormSessionExpiredModal',
+								title: Liferay.Language.get(
+									'your-session-has-expired'
+								),
+							});
+						}
+						else {
+							window.location.reload();
```

Comment:
@mlan0, why we're not letting guest users know about the session expiration as we do for the logged user?

The guest user can be confused with a refresh without any warning, what do you think?